wordpress-mobile/WordPress-iOS
# Posts: Add Remote Search Results

Adds #5265 

This PR adds support for pulling down remote posts that may match the search text within Blog Posts and Pages views.

This adds a new helper `WPContentSearchHelper.swift` that handles the deferred timing/cancellation of searches while typing, so that particular requests aren't made with every key tap while others can.

The UI is still a bit funky here but I think we may need a larger rethink here overall in the future. Ultimately, there are too many states present between the no results view, footer view and spinning indicator.

To test:

1. Open up the Blog Posts for a site that has more posts than what is initially synced.
2. Search for a post that is known to not have already been synced locally and only exists on the remote.
3. The matching post should load upon search via the additional network request.
4. Open the Pages for a site and repeat 1-3. You might change the const `PostServiceDefaultNumberToSync` to something small so that all of a site's Pages aren't loaded on first load.

Test:

1. Open up the Blog Posts for a site with just a few posts, less than the default sync count.
2. Upon searching, no additional network request should be made as they are unneeded.

Needs review: @aerych would you give this a spin? 💃
